### PR TITLE
Detect background changes to database file.

### PR DESCRIPTION
**Pull request by TheZ3ro**
Merged from https://github.com/daniellandau/keepassx/commit/ce4abdc0ef92e3fa2c0054c5c181b7b0575bbcba
Issue #22

### CODE REVIEW

**Comment by TheZ3ro** [comment]
Yes, It's a lot of new code. There is also a change to the search function.

File: src/gui/DatabaseTabWidget.cpp [hunk 1]
```diff
@@ -226,6 +236,114 @@ void DatabaseTabWidget::importKeePass1Database()
...
+{
+    QSet<QString> changedFiles;
+
+    changedFiles = m_changedFiles.subtract(m_expectedFileChanges);
+    m_changedFiles.clear();
+
+    if (changedFiles.isEmpty())
+        return;
+
+    Q_FOREACH (DatabaseManagerStruct dbStruct, m_dbList) {
```
Comment (TheZ3ro) [inline src/gui/DatabaseTabWidget.cpp]: Convert Q_FOREACH loops to C++11 for loops.
Q_FOREACH will de deprecated soon.

**Comment by TheZ3ro** [comment]
I don't know if this is useful, when a database is locked and reopened it will [reload the database file](https://github.com/keepassxreboot/keepassx/blob/develop/src/gui/DatabaseOpenWidget.cpp#L96) so this is useful only if you leave your database open

**Comment by daniellandau** [comment]
Not much I can say, it's not my code. In any case yes: this is exactly for the people (like me) who keep their database open all the time.

**Comment by TheZ3ro** [comment]
Sorry I thought It was your code 😅  
Ok, so It's useful. We must check if the use of the CompositeKey is not a Security Issue

**Pull request merged**